packages/create-pantheon-decoupled-kit/src/templates/next-drupal-app/middleware.ts

```diff
@@ -0,0 +1,42 @@
+import { NextResponse } from 'next/server'
+import type { NextRequest } from 'next/server'
+
+import { i18n } from './i18n-config'
+
+import { match as matchLocale } from '@formatjs/intl-localematcher'
+import Negotiator from 'negotiator'
+
+function getLocale(request: NextRequest): string | undefined {
+  // Negotiator expects plain object so we need to transform headers
+  const negotiatorHeaders: Record<string, string> = {}
+  request.headers.forEach((value, key) => (negotiatorHeaders[key] = value))
+
+  // Use negotiator and intl-localematcher to get best locale
+  let languages = new Negotiator({ headers: negotiatorHeaders }).languages()
+  // @ts-ignore locales are readonly
+  const locales: string[] = i18n.locales
+  return matchLocale(languages, locales, i18n.defaultLocale)
+}
+
+export function middleware(request: NextRequest) {
```

Comment:
The app-dir-i18n-routing example has some commented out code that we might need:
https://github.com/vercel/next.js/blob/canary/examples/app-dir-i18n-routing/middleware.ts#L24
I think we need to ignore our `/public` files here.
Also, I noticed the `favicon.ico` in the `app/[lang]` dir, is that correct?